src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1364,15 +1363,31 @@ IAssemblySymbol computeContainingAssembly(ISymbol s)
                         case SymbolKind.Discard:
                             s = ((IDiscardSymbol)s).Type;
                             continue;
+                        case SymbolKind.FunctionPointer:
+                            var funcPtr = (IFunctionPointerTypeSymbol)s;
+                            if (!isContainingAssemblyInReferences(funcPtr.Signature.ReturnType))
+                            {
+                                return false;
+                            }
+
+                            foreach (var param in funcPtr.Signature.Parameters)
+                            {
+                                if (!isContainingAssemblyInReferences(param.Type))
+                                {
+                                    return false;
+                                }
+                            }
+
+                            return true;
                         case SymbolKind.DynamicType:
                         case SymbolKind.ErrorType:
                         case SymbolKind.Preprocessing:
                         case SymbolKind.Namespace:
                             // these symbols are not restricted in where they can be accessed, so unless they report
                             // a containing assembly, we treat them as in the current assembly for access purposes
-                            return s.ContainingAssembly ?? this.Assembly;
+                            return assemblyIsInReferences(s.ContainingAssembly ?? this.Assembly);
                         default:
-                            return s.ContainingAssembly;
+                            return assemblyIsInReferences(s.ContainingAssembly);
```

Comment:
>return assemblyIsInReferences(s.ContainingAssembly); [](start = 28, length = 52)

It looks like for a NamedType which isn't a definition we should check all type arguments, including from containing types. Could be a breaking change though.

Reply:
Filed [https://github.com/dotnet/roslyn/issues/44345](https://github.com/dotnet/roslyn/issues/44345) to follow up on this.

---
In reply to: [426081997](https://github.com/dotnet/roslyn/pull/44257#discussion_r426081997) [](ancestors = 426081997)